pytato/analysis/__init__.py

```diff
@@ -626,7 +626,7 @@ def rec(self, expr: ArrayOrNames) -> int:
         try:
             return self._cache.retrieve(expr, key=key)
         except KeyError:
-            s = super().rec(expr)
+            s = Mapper.rec(self, expr)
```

Comment:
```suggestion
            # intentionally going to Mapper instead of super() to avoid
            # double caching, see https://github.com/inducer/pytato/pull/585
            s = Mapper.rec(self, expr)
```
(here and elsewhere)